app/controllers/buyers/applications_controller.rb

```diff
@@ -191,8 +188,12 @@ def find_provider
     @provider = current_account
   end
 
-  def find_application_plan
-    @application_plan = @provider.application_plans.find params[:cinstance][:plan_id]
+  def accessible_services
+    current_user.accessible_services.includes(:application_plans)
```

Comment:
I think this is worth memoizing. `#index` will invoke it twice, for example. Plus, by doing this, you can save a couple lines of changes in the `#index` action too.

Reply:
Well it is a Relation, memoizing or not does not change it as it has not been queried yet.
But yes good remark